### PR TITLE
adding support for patterns in summary report

### PR DESCRIPTION
**Pull request by fikin**
hope you're open for minor improvements?

this adds support for patterns via -l flag to summary report (-s).

### CODE REVIEW

File: src/luacov-console.lua [hunk 1]
```diff
@@ -27,6 +27,14 @@ local function colored_print(color, text)
     print(start_color .. text .. end_color)
 end
 
+local function colored_print_if_pattern(color, text, patterns)
```
Comment (spacewander) [inline src/luacov-console.lua:30]: According to this new name, maybe we should introduce an optimization that if the patterns are nil, we call colored_print directly. And for cases without patterns, we can simply pass a nil as the patterns.

Comment (fikin) [reply src/luacov-console.lua:30]: it is an optimisation true and it passed my mind as well. 
but then thought to go with less cyclomatic complexity instead.
do you really want that if-statement?

Comment (spacewander) [reply src/luacov-console.lua:30]: I think adding the if-statement is better as in most cases we don't pass a pattern to this tool.

Comment (fikin) [reply src/luacov-console.lua:30]: ok, here you go

**Review by spacewander — APPROVED** [review]

**Comment by spacewander** [comment]
@fikin 
Merged. Thanks!
Should I kick off a new release?

**Pull request merged**